### PR TITLE
Add support for "abs-capture-time" extension. 

### PR DESCRIPTION
**Pull request by boks1971**
Currently, it is just passed through from publisher -> subscriber side.

@livekit/sfu-devs : Not going to merge just yet as it is unclear what the support for `abs-capture-time` is, but have some other minor clean up (mostly moving files and grouping them better). So, want to get 👀 on this. May remove the `abs-capture-time` bits before merging, but I think the clean up is still useful.

### CODE REVIEW

File: pkg/rtc/mediaengine.go [hunk 1]
```diff
@@ -29,9 +29,22 @@ const (
...
+var redCodecCapability = webrtc.RTPCodecCapability{
+	MimeType:    sfu.MimeTypeAudioRed,
+	ClockRate:   48000,
+	Channels:    2,
+	SDPFmtpLine: "111/111",
+}
+var videoRTX = webrtc.RTPCodecCapability{
+	MimeType:  videoRTXMimeType,
+	ClockRate: 90000,
+}
```
Comment (boks1971) [inline pkg/rtc/mediaengine.go:47]: changing it for easier reading.

File: pkg/rtc/mediatrackreceiver.go [hunk 2]
```diff
@@ -34,7 +34,7 @@ import (
 	"github.com/livekit/livekit-server/pkg/rtc/types"
 	"github.com/livekit/livekit-server/pkg/sfu"
 	"github.com/livekit/livekit-server/pkg/sfu/buffer"
-	"github.com/livekit/livekit-server/pkg/sfu/dependencydescriptor"
+	"github.com/livekit/livekit-server/pkg/sfu/rtpextension/dependencydescriptor"
```
Comment (boks1971) [inline pkg/rtc/mediatrackreceiver.go:37]: most of the changes are grouping all extension stuff under `sfu/rtpextension` directory. So, a few things moved and that is most of the changes.

File: pkg/sfu/buffer/buffer.go [hunk 3]
```diff
@@ -756,6 +766,11 @@ func (b *Buffer) getExtPacket(rtpPacket *rtp.Packet, arrivalTime time.Time, flow
 		}
 	}
 
+	if b.absCaptureTimeExtID != 0 {
+		ep.AbsCaptureTimeExt = rtpPacket.GetExtension(b.absCaptureTimeExtID)
+		b.logger.Debugw("ACT DEBUG", "size", len(ep.AbsCaptureTimeExt)) // REMOVE
```
Comment (boks1971) [inline pkg/sfu/buffer/buffer.go:771]: Will remove after debugging.

File: pkg/sfu/downtrack.go [hunk 4]
```diff
@@ -730,13 +742,46 @@ func (d *DownTrack) WriteRTP(extPkt *buffer.ExtPacket, layer int32) error {
...
+				},
+			)
+
+			// NOTE: play out delay extension is not cached in sequencer,
+			// i. e. they will not be added to retransmitted packet.
+			// But, it is okay as the extension is added till a RTCP Receiver Report for
+			// the corresponding sequence number is received.
+			// The extreme case is all packets containing the play out delay are lost and
+			// all of them retransmitted and an RTCP Receiver Report received for those
+			// retransmited sequence numbers. But, that is highly improbable, if not impossible.
```
Comment (boks1971) [inline pkg/sfu/downtrack.go:769]: adding notes about design decision to not include this in sequencer.

**Review by cnderrauber — APPROVED** [review]

**Comment by boks1971** [comment]
@livekit/sfu-devs Going to merge this. Have the `abs-capture-time` bits in there, but commenting out registering that extension from media engine. So, it won't be negotiated. We can enable it if/when we need it. Or can delete it if we feel like it will not be used.

**Pull request merged**